test/models/work_test.rb

```diff
@@ -0,0 +1,369 @@
+        expect(albums[0].title).must_equal works(:album_ten).title
+      end
+
+      it "returns empty array if there are no albums" do 
+        albums = Work.sort_album_works
+        albums.each do |a|
+          a.category = "book"
+          a.save
+        end
+
+        result = Work.top_albums
+        expect(result).must_equal []
+      end
+
+      it "returns 5 albums if there are only 5 of them" do 
+        albums = Work.sort_album_works
+        albums.each_with_index do |album, index|
+          album.category = "book"
+          album.save
+          break if index == 4
+        end
+
+        result = Work.top_albums.count
+        puts "RESULT = #{result}"
+        expect(result).must_equal 5
+      end
+    end
+
+    describe "top_books" do 
+      it "returns 10 books" do 
+        books = Work.top_books
+
+        result = books.length
+        expect(result).must_equal 10
+
+        vote = Vote.create(
+          user_id: users(:user_two).id, 
+          work_id: works(:dandelion_wine).id,
+          date: "Today"
+        )
+
+        num_of_votes = works(:dandelion_wine).votes.count 
+        expect(num_of_votes).must_equal 1
+
+        works = Work.top_books
+        expect(works[0].title).must_equal works(:dandelion_wine).title
+      end
+
+      it "returns empty array if there are no books" do 
+        books = Work.sort_book_works
+        books.each do |b|
+          b.category = "movie"
+          b.save
+        end
+
+        result = Work.top_books
+        expect(result).must_equal []
+      end
+
+      it "returns 5 books if there are only 5 of them" do 
+        books = Work.sort_book_works
+        books.each_with_index do |book, index|
+          book.category = "movie"
+          book.save
+          break if index == 4
+        end
+
+        result = Work.top_books.count
+        puts "RESULT = #{result}"
+        expect(result).must_equal 5
+      end  
+    end
+
+    describe "top_movies" do
+      it "returns 10 movies" do 
+        movies = Work.top_movies
+
+        result = movies.length
+        expect(result).must_equal 10
+
+        vote1 = Vote.create(
+          user_id: users(:user_two).id, 
+          work_id: works(:titanic).id,
+          date: "Today"
+        )
+
+        vote2 = Vote.create(
+          user_id: users(:user_one).id, 
+          work_id: works(:movie_two).id,
+          date: "Today"
+        )
+
+        vote3 = Vote.create(
+          user_id: users(:user_two).id, 
+          work_id: works(:movie_two).id,
+          date: "Today"
+        )
+
+        num_of_votes = works(:titanic).votes.count 
+        expect(num_of_votes).must_equal 1
+
+        works = Work.top_movies
+        expect(works[0].title).must_equal works(:movie_two).title
+      end
+
+      it "returns empty array if there are no movies" do 
+        movies = Work.sort_movie_works
+        movies.each do |m|
+          m.category = "book"
+          m.save
+        end
```

Comment:
Instead of reassigning all the categories here you could simply do `Work.destroy_all`  Or you could do.

`Work.where(category: 'movie').destroy_all` to just destroy all the movies.